File: lib/scala/generic/src/main/scala/org/paradicms/lib/generic/models/domain/Image.scala
package org.paradicms.lib.generic.models.domain

import io.lemonlabs.uri.Url

final case class Image(
                        exactDimensions: Option[ImageDimensions],
                        maxDimensions: Option[ImageDimensions],
                        url: Url
                      ) {
  val uri = url
}

File: lib/ts/models/src/Image.ts
import {ImageDimensions} from "./ImageDimensions";
import {Rights} from "./Rights";

export interface Image {
  readonly depictsUri: string;
  readonly exactDimensions?: ImageDimensions;
  readonly institutionUri: string;
  readonly maxDimensions?: ImageDimensions;
  readonly originalImageUri?: string;
  readonly rights?: Rights;
  readonly uri: string;
}
